### ISSUE TITLE
Batched seeds does not work for `StableDiffusionControlNetPipeline` (and Img2Img pipeline)

### ISSUE THREAD

**Issue by AmitMY** [huggingface/diffusers#6218]
### Describe the bug

When trying to create images with a consistent seed in a batch, the images turn out different.

### Reproduction

Load the model and pipeline
```py
import torch
from diffusers import StableDiffusionControlNetImg2ImgPipeline, ControlNetModel, UniPCMultistepScheduler

controlnet = ControlNetModel.from_pretrained("sign/signwriting-illustration", torch_dtype=torch.float16)
pipe = StableDiffusionControlNetImg2ImgPipeline.from_pretrained(
    "runwayml/stable-diffusion-v1-5", 
    controlnet=controlnet, 
    safety_checker=None, 
    torch_dtype=torch.float16)

# Instead of using Stable Diffusion's default PNDMScheduler, we use one of the currently fastest diffusion model schedulers, called UniPCMultistepScheduler. 
pipe.scheduler = UniPCMultistepScheduler.from_config(pipe.scheduler.config)

# the pipeline automatically takes care of GPU memory management.
pipe.enable_model_cpu_offload()

# attention layer acceleration
pipe.enable_xformers_memory_efficient_attention()
```

Run batch inference:
```py
img2img_init = Image.new('RGB', (512, 512), 'white')

batch=4

prompt = "An illustration of a man with short hair, with black arrows."
output = pipe(
    prompt=[prompt]*batch,
    image=[img2img_init]*batch,
    control_image=[img2img_init]*batch,
    negative_prompt=["colored, lowres, bad anatomy, worst quality, low quality, watermark"]*batch,
    generator=[torch.manual_seed(1)]*batch,
    num_inference_steps=20,
    strength=1.0,
    controlnet_conditioning_scale=1.0,
    guidance_scale=7.5,
)

image_grid(output.images, 2, 2)
```

Expecting the output to be the same image four times. Instead, getting this:

![image](https://github.com/huggingface/diffusers/assets/5757359/735fc196-c0d2-4483-a4a8-f83c08cefa26)

### Logs

```shell
no logs.
```

### System Info

- `diffusers` version: 0.24.0
- Platform: Linux-6.1.58+-x86_64-with-glibc2.35
- Python version: 3.10.12
- PyTorch version (GPU?): 2.1.2+cu121 (True)
- Huggingface_hub version: 0.19.4
- Transformers version: 4.35.2
- Accelerate version: 0.25.0
- xFormers version: 0.0.23.post1
- Using GPU in script?: YES
- Using distributed or parallel set-up in script?: NO

### Who can help?

This is about ControlNet: @sayakpaul @yiyixuxu

**Comment by sayakpaul** [comment]
Can we reproduce it for the case when the batch size is 1?

**Comment by AmitMY** [comment]
batch size 1 - the image is always the same

**Comment by sayakpaul** [comment]
@yiyixuxu would you be able to take a look?

**Comment by patrickvonplaten** [comment]
Hey @AmitMY, 

Can you disable xformers to see if this might be the culprit? We have tests that verify that the batched seed works for this pipeline

**Comment by AmitMY** [comment]
Without `xformers` the result is the same (I removed `pipe.enable_xformers_memory_efficient_attention()`)

See colab for repro https://colab.research.google.com/drive/1y4aEtGrSeZHgklxs82URtZk_5USM5X_I?usp=sharing

**Comment by yiyixuxu** [comment]
Note that if you do `[torch.Generator().manual_seed(seed)] * 4`; you will get a list of 4 "generators" points to the same underlying tensor object. Even though it looks like we passed a list of 4 generators of seed 0, we actually only passed one, which will be used 4 times in a sequential manner to generate the batched initial latents; 

                [<torch._C.Generator at 0x796a8934c7f0>,
                 <torch._C.Generator at 0x796a8934c7f0>,
                 <torch._C.Generator at 0x796a8934c7f0>,
                 <torch._C.Generator at 0x796a8934c7f0>]


The correct way to create batched seeds is
 
```python
seed = 0
generators = [torch.Generator().manual_seed(seed) for _ in range(batch)]
```

This gives you a unique generator object for each image in the batch :) and you will get 4 same images as results

```
[<torch._C.Generator at 0x796a8934c990>,
 <torch._C.Generator at 0x796a8934d1f0>,
 <torch._C.Generator at 0x796a8934d0d0>,
 <torch._C.Generator at 0x796a8934c610>]

```

cc @patrickvonplaten @stevhliu , is this the expected behavior for batched seeds? (pretty unintuitive I have to say) should we talk about it in the doc?

**Comment by AmitMY** [comment]
Thanks!
Even more interesting, is that doing
```py
[torch.manual_seed(1) for _ in range(batch)]
```
does not work, while doing
```py
[torch.Generator().manual_seed(1) for _ in range(batch)]
```

I don't know why.
Feel free to close, or maybe give a warning if the seed is the same generator in an array

**Comment by gtebbutt** [comment]
@AmitMY that'll likely be `torch.manual_seed` returning a reference to the same generator each time - from [the source](https://pytorch.org/docs/stable/_modules/torch/random.html#manual_seed) it looks like that function always returns a reference to `default_generator`, which handles torch's global random state, whereas `torch.Generator().manual_seed` is setting the seed on a newly created generator instance each time.

**Comment by stevhliu** [comment]
> is this the expected behavior for batched seeds? (pretty unintuitive I have to say) should we talk about it in the doc?

I can add a quick note/explanation [here](https://huggingface.co/docs/diffusers/using-diffusers/reusing_seeds) @yiyixuxu, wdyt?

**Comment by yiyixuxu** [comment]
@stevhliu sounds good! thank you!

**Comment by stevhliu** [comment]
Completed in #6237
